tests/integration/tier0/non-destructive/eus/test_eus_support.py

```diff
@@ -0,0 +1,102 @@
+    # System not recognized as an EUS candidate with --eus option used
+    ("--eus", REGULAR_REPOID_MSG, "", False, False),
+    # System recognized as and EUS candidate before the start of EUS phase without --eus option
+    ("", REGULAR_REPOID_MSG, EUS_SYS_PRE_EUS_PHASE, False, False),
+    # System recognized as and EUS candidate before the start of EUS phase with --eus option
+    ("--eus", EUS_REPOID_MSG, EUS_SYS_PRE_EUS_PHASE, True, False),
+    # System recognized as and EUS candidate after the start of EUS phase without --eus option
+    ("", REGULAR_REPOID_MSG, EUS_SYS_EUS_PHASE, False, True),
+]
+
+
+@pytest.mark.parametrize(
+    "additional_options, repoid_message, modified_mapping, is_eus, eus_recommend",
+    eus_support_parameters,
+    ids=["non-eus-sys-eus-opt", "eus-sys-pre-eus-no-opt", "eus-sys-pre-eus-eus-opt", "eus-sys-eus-phase-no-opt"],
+)
+@pytest.mark.test_eus_support
+def test_eus_support(
+    convert2rhel, eus_mapping_update, additional_options, repoid_message, modified_mapping, is_eus, eus_recommend
+):
+    """
+    Test verifying correct behavior when converting EUS candidates.
+    EUS_MINOR_VERISONS mapping in convert2rhel/systeminfo.py is modified to mock the different scenarios.
+    Verified scenarios (handled by patest parametrization):
+    1/ The running minor version is removed from the mapping to not be considered an EUS candidate.
+        The --eus option is used in the command and enabling regular (non-eus) repoids is verified.
+    2/ The date of the start of the EUS phase is set far to the future to simulate the EUS phase
+        did not start yet.
+        The --eus option *is not used* and regular repoids enablement is expected.
+    3/ The date of the start of the EUS phase is set far to the future to simulate the EUS phase
+        did not start yet.
+        The --eus option *is used* and EUS repoids enablement is expected.
+    4/ The date of the start of the EUS phase is set far to the past to simulate the EUS phase began.
+        The --eus option *is used*.
+        The report is expected to print out a WARNING and advise to use the --eus option.
+    """
+    eus_mapping_update(modified_mapping)
+    with convert2rhel(
+        "analyze -y --debug --no-rpm-va --serverurl {} -u {} -p {} {}".format(
+            env.str("RHSM_SERVER_URL"), env.str("RHSM_USERNAME"), env.str("RHSM_PASSWORD"), additional_options
+        )
+    ) as c2r:
+        # If the system is an EUS candidate and current date is past the beginning of the EUS phase
+        # we set the eus_recommend parameter to True
+        # We expect the report to print out a warning to use --eus option
+        if eus_recommend:
+            c2r.expect(repoid_message, timeout=120)
+            c2r.expect("EUS_SYSTEM_CHECK::EUS_COMMAND_LINE_OPTION_UNUSED")
+        else:
+            c2r.expect(repoid_message, timeout=120)
+            # In case the EUS repositories should get enabled, we set is_eus parameter to True
```

Comment:
I would remove this comment as it is not giving any different information then the code itself.
Or rephrase it to:
```suggestion
            # If  is_eus is True, expect a corresponding EUS message displayed
```